coretex/cli/resources/update_script_template.sh

```diff
@@ -0,0 +1,136 @@
+        echo "Checking node version..."
+        # Get latest image digest from docker hub
+        manifest_output=$($DOCKER_PATH manifest inspect $REPOSITORY:$TAG --verbose)
+
+        # get digest from docker hub
+        digest_line=$(echo "$manifest_output" | grep -o '"digest": ".*"' | head -n 1)
+        latest_digest=$(echo "$digest_line" | cut -d '"' -f 4)
+
+        # get digest from local container 
+        current_digest=$($DOCKER_PATH image ls --digests | grep $REPOSITORY | grep $TAG | awk '{{print $3}}')
+        current_digest=$(echo "$current_digest" | awk '{{$1=$1;print}}')
+
+        # Compare digests
+        if [[ $latest_digest != $current_digest ]]; then
+            return 0 # Return True since there is new image to be pulled from docker hub
+        else
+            echo "Node version is up to date."
+            return 1
+        fi
+    else
+        echo "Node is not active."
+        return 1
+    fi
+}}
+
+function pull_image {{
+    if $DOCKER_PATH image pull "$IMAGE"; then
+        echo "Image pulled successfully: $IMAGE"
+        return 0
+    else
+        echo "Failed to pull image: $IMAGE"
+        return 1
+    fi
+}}
+
+# Define function to stop and remove the container
+function stop_node {{
+    local api_response=$(curl -s "$NODE_STATUS_ENDPOINT")
+    local status=$(echo "$api_response" | sed -n 's/.*"status":\([^,}}]*\).*/\1/p')
+
+    if [ "$status" -eq 3 ]; then
+        echo "Node is busy, stopping node update."
+        return 1
+    fi
+
+    echo "Stopping and removing the container: $CONTAINER_NAME"
+    $DOCKER_PATH stop "$CONTAINER_NAME" && $DOCKER_PATH rm "$CONTAINER_NAME"
+
+    if $DOCKER_PATH network inspect $NETWORK_NAME; then
+        echo "Removing the network: $NETWORK_NAME"
+        $DOCKER_PATH network rm "$NETWORK_NAME"
+    fi
+
+    echo "Node successfully stopped."
+}}
+
+# Define function to start the node with the latest image
+function start_node {{
+    if $DOCKER_PATH network inspect $NETWORK_NAME; then
+        echo "Removing the network: $NETWORK_NAME"
+        $DOCKER_PATH network rm "$NETWORK_NAME"
+    fi
+
+    echo "Creating network"
+    $DOCKER_PATH network create --driver bridge $NETWORK_NAME
+
+    echo "Starting the node with the latest image"
+    $DOCKER_PATH run -d --env "serverUrl=$SERVER_URL" --env "storagePath=$STORAGE_PATH" --env "nodeAccessToken=$NODE_ACCESS_TOKEN" --restart $RESTART_POLICY -p $PORTS --cap-add $CAP_ADD --network "$NETWORK_NAME" --memory $RAM_MEMORY --memory-swap $SWAP_MEMORY --shm-size $SHARED_MEMORY --name "$CONTAINER_NAME" "$IMAGE"
+}}
+
+# Define function to update node
+function update_node {{
+    echo "Updating node"
+    if should_update; then
+        if pull_image; then
+            if stop_node; then
+                start_node
+            fi
+        fi
+    fi
+}}
+
+# Function to run node update
+function run_node_update {{
+    echo "Running command: $NODE_UPDATE_COMMAND"
+    $NODE_UPDATE_COMMAND
```

Comment:
Is `NODE_UPDATE_COMMAND` necessary or can it be replaced with a direct function call?